src/services/feedProcessors/MultiFeedProcessor.ts

```diff
@@ -1,56 +1,103 @@
 
     const promiseMap = {
-      CryptoCompareMultiProcessor: () => this.cryptoCompareMultiProcessor.apply(feedFetchers),
-      CoingeckoMultiProcessor: () => this.coingeckoMultiProcessor.apply(feedFetchers),
-      UniswapV3MultiProcessor: () => this.uniswapV3MultiProcessor.apply(feedFetchers),
-      SovrynMultiProcessor: () => this.sovrynMultiProcessor.apply(feedFetchers),
-      ByBitMultiProcessor: () => this.byBitMultiProcessor.apply(feedFetchers),
-      BinanceMultiProcessor: () => this.binanceMultiProcessor.apply(feedFetchers),
+      [FetcherName.BY_BIT]: () =>
+        this.byBitSpotFetcher.apply(this.getInputs<InputParamsByBit>(feedFetchers, FetcherName.BY_BIT)),
+      [FetcherName.BINANCE]: () =>
+        this.binancePriceMultiFetcher.apply(this.getInputs<InputParamsBinance>(feedFetchers, FetcherName.BINANCE)),
+      [FetcherName.COINGECKO_PRICE]: () =>
+        this.coingeckoPriceMultiFetcher.apply(
+          this.getInputs<InputParamsCoingecko>(feedFetchers, FetcherName.COINGECKO_PRICE),
+        ),
+      [FetcherName.UNISWAP_V3]: () =>
+        this.uniswapV3MultiFetcher.apply(this.getInputs<InputParamsUniswapV3>(feedFetchers, FetcherName.UNISWAP_V3)),
+      [FetcherName.SOVRYN_PRICE]: () =>
+        this.sovrynFetcher.apply(this.getInputs<InputParamsSovryn>(feedFetchers, FetcherName.SOVRYN_PRICE)),
     };
 
     const promises = Object.values(promiseMap).map((fn) => fn());
     const classNames = Object.keys(promiseMap);
 
     const promisesResults = await Promise.allSettled(promises);
 
+    let response: unknown[] = [];
+    const payloads: PriceDataPayload[] = [];
+
     promisesResults.forEach((result, i) => {
       if (result.status === 'fulfilled') {
-        response = mergeArrays(response, result.value);
+        const symbols = feedFetchers
+          .filter((fetcher) => fetcher.name == classNames[i])
+          .map((fetcher) => fetcher.symbol);
+
+        const timestamp = result.value.timestamp || this.timeService.apply();
+
+        for (const [ix, output] of result.value.prices.entries()) {
+          if (!output) continue;
+
+          const result = this.feedSymbolChecker.apply(symbols[ix]);
+          if (!result) continue;
+
+          const [feedBase, feedQuote] = result;
+
+          payloads.push({
+            fetcher: classNames[i],
+            value: output.toString(),
+            valueType: PriceValueType.Price,
+            timestamp,
+            feedBase,
+            feedQuote,
+            fetcherSource: '',
+          });
+        }
+
+        response = response.concat(result.value.prices);
         this.logger.debug(`[MultiFeedProcessor] fulfilled ${classNames[i]}: ${JSON.stringify(result.value)}`);
       } else {
         this.logger.warn(`[MultiFeedProcessor] Ignored ${classNames[i]}. Reason: ${result.reason}`);
       }
     });
 
+    await this.priceDataRepository.savePrices(payloads);
```

Comment:
this is now what we want, we want each fetcher to save price because we need this for aggregation functionality.

Reply:
addressed. I added an interface for the multifetchers that accepts symbols as options and now we can call them inside the multi fetchers.